### PR TITLE
feat: prompt for push notification permission

### PR DESCRIPTION
**Pull request by ami-aman**
This PR uses `customerio-reactnative`'s latest method to get push notification authorization status and show prompt for push permission.

Main screen now has 3 buttons 
- Get started
- Show push prompt
- Get push permission status

**Show push prompt**
Assume that the app has not been authorized yet, you will see a prompt seeking user's consent for push notifications.
In case, the app has already been granted/denied permissions (or any other status), then you'll be shown an alert with current status.

**Get push permission status**
This will simply get the status of push notification authorization status in a callback.

THIS IS A DRAFT PR AS WORK IS IN PROGRESS
IMG_5842

### CODE REVIEW

File: components/FeaturesUpdate.js [hunk 1]
```diff
@@ -47,6 +47,29 @@ const FeaturesUpdate = ({navigation}) => {
...
+    } 
+
+    const getPushPermissionStatus = () => {
+      CustomerIO.getPushPermissionStatus((response) => {
+        performActionOnPushStatus(response)
+      })
+    }
+
+    const requestPushPermissionPrompt = () => {
+      // Todo: change
```
Comment (mrehan27) [inline components/FeaturesUpdate.js:62]: Is there still anything pending or should we remove this comment now?

File: ios/AppDelegate.swift [hunk 2]
```diff
@@ -40,7 +40,7 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
     // firebase used for app distribution.
     // FirebaseApp.configure(options: options)
     
-    registerForPushNotifications()
+//    registerForPushNotifications()
```
Comment (mrehan27) [inline ios/AppDelegate.swift:43]: May be we should remove this completely? 🤔

**Review by mrehan27 — APPROVED** [review]
Just minor queries, rest looks good to me 👍🏻

File: components/FeaturesUpdate.js (hunk 1 above)
Comment (ami-aman) [reply components/FeaturesUpdate.js:110]: Oops ! Thanks for reminding

File: ios/AppDelegate.swift (hunk 2 above)
Comment (ami-aman) [reply ios/AppDelegate.swift]: Yes, removed now.

**Pull request merged**